pgml-dashboard/src/utils/markdown.rs

```diff
@@ -1,45 +1,44 @@
 use crate::{templates::docs::TocLink, utils::config};
-use comrak::{
-    adapters::{HeadingAdapter, HeadingMeta, SyntaxHighlighterAdapter},
-    arena_tree::Node,
-    nodes::{Ast, AstNode, NodeValue},
-    parse_document, Arena, ComrakExtensionOptions, ComrakOptions, ComrakRenderOptions,
-};
+
 use std::cell::RefCell;
 use std::collections::{HashMap, HashSet};
 use std::path::{Path, PathBuf};
 use std::sync::Arc;
 
 use aho_corasick::{AhoCorasick, AhoCorasickBuilder, MatchKind};
+use anyhow::Result;
+use comrak::{
+    adapters::{HeadingAdapter, HeadingMeta, SyntaxHighlighterAdapter},
+    arena_tree::Node,
+    nodes::{Ast, AstNode, NodeValue},
+    parse_document, Arena, ComrakExtensionOptions, ComrakOptions, ComrakRenderOptions,
+};
+use convert_case::Casing;
 use itertools::Itertools;
 use lazy_static::lazy_static;
 use tantivy::collector::TopDocs;
 use tantivy::query::{QueryParser, RegexQuery};
 use tantivy::schema::*;
 use tantivy::tokenizer::{LowerCaser, NgramTokenizer, TextAnalyzer};
 use tantivy::{Index, IndexReader, SnippetGenerator};
+use url::Url;
 
+use crate::templates::docs::NavLink;
 use std::fmt;
-use std::sync::atomic::AtomicUsize;
 
 pub struct MarkdownHeadings {
-    header_counter: AtomicUsize,
 }
 
 impl MarkdownHeadings {
     pub fn new() -> Self {
         Self {
-            header_counter: AtomicUsize::new(0),
         }
     }
 }
 
 impl HeadingAdapter for MarkdownHeadings {
     fn enter(&self, meta: &HeadingMeta) -> String {
-        let id = self
-            .header_counter
-            .fetch_add(1, std::sync::atomic::Ordering::SeqCst);
-        let id = format!("header-{}", id);
+        let id = meta.content.to_case(convert_case::Case::Kebab);
```

Comment:
Not unique enough, we have duplicate headers in blogs and docs.

Reply:
We should dedup them. URLS are part of the UX, and we're also losing control of this to gitbook.